### ISSUE TITLE
Tiling setting

### ISSUE THREAD

**Issue by SirNate0** [thygate/stable-diffusion-webui-depthmap-script#444]
Is it possible to have the script honor the Tiling setting for the image? The image itself seems to tile very nicely. However, the depth image has a distinct curve to it that does not tile. Using a high-pass filter after the fact mostly fixes it, but it is not perfect, leaving a slight seam.

![00003-2024-05-29_11-46-56 (best quality), (high quality), stone wall texture, gray 3835511387](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/10181610/3062f116-ddaf-4bb3-8be0-95d7ffa3a93f)
![00004-2024-05-29_11-47-15 (best quality), (high quality), stone wall texture, gray 3835511387](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/10181610/7ae85a00-e618-429e-b668-fd57943fb4a3)
With high-pass filter (and offset by 1/2 to show the seam)
![Offset-Krita-Gaussian-High-Pass--00004-2024-05-29_11-47-15 (best quality), (high quality), stone wall texture, gray 3835511387](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/10181610/0cea82cd-7f44-46eb-95a8-d86b60106718)

**Comment by SirNate0** [comment]
While not seamless, if I turn off "BOOST (multi-resolution merging)" and turn on "Match net size to input size", the result does not have the broad sloping depth to the corner.
![depthmap-0004-offset](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/10181610/f12a5b9d-3fcf-48db-8d2b-ff8395159135)

**Comment by SirNate0** [comment]
To actually make it seamless, it's possible to use the same approach as sd_hijack.py and simply force all the Conv2d layers to have the circular padding mode. E.g. adding this to `load_models()` in depthmap_generation.py, though I've only tested it on the res101 model (and it failed with BOOST enabled - probably have to make the same adjustment to the BOOST model, though it might be more complicated since I think it is breaking the image into different patches...)
```py
    def apply_circular(enable):
        def flatten(el):
            flattened = [flatten(children) for children in el.children()]
            res = [el]
            for c in flattened:
                res += c
            return res
        layers = flatten(model)
        for layer in [layer for layer in layers if type(layer) == torch.nn.Conv2d]:
            layer.padding_mode = 'circular' if enable else 'zeros'
    apply_circular(True)
```

**Comment by semjon00** [comment]
Wow, this is impressive! I didn't even think of that, it indeed works without finetuning!

**Comment by semjon00** [comment]
I wrote code that adds a new checkbox for the tiling mode, but my results are somewhat lacking...
No tiling:
![image](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/54073010/d7279295-2cf7-4240-9754-ba6a70aad687)
Yes tiling:
![image](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/54073010/31b8a0b7-a760-45b3-811e-620dc7b71505)

Four tiles on an image, I still see some seam. Should I use high-pass filter?

Depth Anything, match net size to input size, but same with other models and sizes

What could the problem be?

**Comment by semjon00** [comment]
Ok, this feature has just hit the main branch. It still has some seams... It would be awesome if you could take a look and identify the reason...

**Comment by semjon00** [comment]
@SirNate0
